civo/main.tf

```diff
@@ -0,0 +1,32 @@
+resource "civo_network" "demo" {
+  label = "demo"
+}
+
+resource "civo_firewall" "demo" {
+  name                 = "demo"
+  network_id           = civo_network.demo.id
+  create_default_rules = false
+
+  ingress_rule {
+    protocol   = "tcp"
+    port_range = "6443"
+    cidr       = ["0.0.0.0/0"]
+    label      = "kubernetes-api-server"
+    action     = "allow"
+  }
+}
+
+resource "civo_kubernetes_cluster" "demo" {
+  name               = "demo"
+  firewall_id        = civo_firewall.demo.id
+  network_id         = civo_network.demo.id
+  cluster_type       = "talos"
+  cni                = "cilium"
```

Comment:
This config is not honored by the API, the created cluster has Flannel installed. 

Cilium is required for advanced features like Network Policies. I have reached out to the Civo support team.

Reply:
I heard back from Civo. `cilium` is only supported with `k3s` at the moment.